### PR TITLE
Prepare server to deploy as a separate package

### PR DESCRIPTION
**Pull request by danielpza**
Installing `eslint-server` would start the eslint standalone server. It will need to be published to the npm repository:

```sh
cd server
npm install
npm publish # or --dry-run to check everything is ok
```

You can try it out by running `npm link` inside the server/ directory, it will link the command to the global path.

You can change the name of the package/command by modifying the `name` property of `server/package.json`, I would suggest `eslint-language-server` or something like that

Address #1003

### CODE REVIEW

**Comment by dbaeumer** [comment]
@danielpza beside publishing it to npm I think we would need a documentation of the private non LSP message the ESLint server uses / provides. Otherwise it will be hard to talk to the server. In addition we need to decide how we deal with versioning if the server uses new private messages.

**Comment by rchl** [comment]
Here is a rundown of custom notifications/requests:

Notifications from the server:
 - `eslint/status`
 - `eslint/exitCalled`

Requests from the server:
 - `eslint/openDoc`
 - `eslint/noConfig`
 - `eslint/noLibrary`
 - `eslint/probeFailed`
 - `eslint/confirmLocalESLint`

<hr>

There is some custom handling needed when the client is responding to `workspace/configuration`:
 - The `workspaceFolder` needs to be set for files that are within the workspace
 - If the client/user has defined `workingDirectories` then this setting needs to resolved into a value that is set as `workingDirectory` property.

<hr>

The `probe` configuration option also needs to be resolved and the resolved value set as `validate` configuration.

This is optional but it provides a better experience if implemented.

<hr>

So basically all of this would need to be documented as it's a functionality supported by the server but currently handled within the extension so it will be missing when using server with another editor.

**Comment by danielpza** [comment]
I included some docs in the PR, however I don't know much about the workings of the server to document them all. I think it is a good start, the other clients have been using the server without any docs so I think it will be fine and further docs can be included later.

**Comment by yyoncho** [comment]
> @danielpza beside publishing it to npm I think we would need a documentation of the private non LSP message the ESLint server uses / provides. Otherwise it will be hard to talk to the server. In addition we need to decide how we deal with versioning if the server uses new private messages.

@dbaeumer thank you for that. I just want to mention that the majority of the servers(with few exceptions) do not document these methods and we(at emacs side) are used to reverse-engineering the extension methods.

**Comment by dbaeumer** [comment]
For documenting the configuration request. What needs to go into the document is not that the request is send. What we need to document is the structure of the response. (e.g. this type https://github.com/microsoft/vscode-eslint/blob/master/server/src/eslintServer.ts#L217)

**Comment by danielpza** [comment]
@dbaeumer I think we should leave the documentation for another PR or for the wiki. We are missing out on the standalone server

**Comment by dbaeumer** [comment]
I disagree. The reason is simple: not having the documentation will result in additional issue that will come via this repository. So I would really like to see the documentation  together with the npm package.

**Comment by danielpza** [comment]
What additional issues do you expect to come to the repository? If they are documentation related it will be a good way to expand the documentation on demand.

**Comment by dbaeumer** [comment]
People asking for the documentation that someone then has to write.

**Comment by Sebastianokunkelo** [comment]
danielpza:prepare-server-deploy

**Comment by williamboman** [comment]
FWIW, this repo is packaged via https://github.com/hrsh7th/vscode-langservers-extracted as of recently. Obviously not an official package, but thought I'd mention it
